packages/tinytest/tinytest_server.js

```diff
@@ -78,15 +78,13 @@ Meteor.methods({
   'tinytest/clearResults'(runId) {
     check(runId, String);
 
-    const handles = handlesForRun.get(runId);
-    // Fix: Exception while invoking method 'tinytest/clearResults' TypeError: Cannot read property 'forEach' of undefined
-    // Second call of the method will always run without exception
-    if (typeof handles !== 'undefined') {
-      handles.forEach(handle => {
-        // XXX this doesn't actually notify the client that it has been
-        // unsubscribed.
-        handle.stop();
-      });
+    if (handlesForRun.has(runId)) {
+      handlesForRun.get(runId).forEach(handle => {
```

Comment:
@sebakerckhof problem is, value `handlesForRun.get(runId)` maybe `undefined`, and you trying using the method `.forEach()` for undefined.

Reply:
`
```suggestion
      const handles = handlesForRun.get(runId);
      // We checking for type, for avoiding `undefined.forEach()`
      if (typeof handles !== 'undefined') {
        handles.forEach(handle => {
```

Reply:
But it would only return undefined if the map doesn't have the key, right? It's initialized as a Set and I don't see any place where this can change except for the delete. Am I overlooking something?

Reply:
@sebakerckhof is some cases we may get this state for variable `handlesForRun`:
```js
const handlesForRun = new Map([
  // Correct run with `set` of handles, `.forEach()` is work
  ['runId1', new Set(handle1, handleN)],
  // Incorrect run with `undefined`, `.forEach()` is doesn't work
  ['runId2', undefined],
]);
```

So, for `runId2`:

```
const runId = 'runId2';

handlesForRun.get(runId).forEach(handle => {
  // ...
});
```

You get the error:

```
Uncaught TypeError: Cannot read property 'forEach' of undefined
```

Reply:
Yes, I understand what you mean. What I'm questioning is how we can get into that situation. Because this would mean there is a `handlesForRun.set(runId2, undefined)` somewhere, which as far as I can tell, should not happen. So there may be a deeper problem if this is the case.